x/interchainqueries/keeper/keeper.go

```diff
@@ -86,6 +86,14 @@ func (k Keeper) SetLastRegisteredQueryKey(ctx sdk.Context, id uint64) {
 func (k Keeper) SaveQuery(ctx sdk.Context, query *types.RegisteredQuery) error {
 	store := ctx.KVStore(k.storeKey)
 
+	// We're doing this because LastSubmittedResultRemoteHeight is a struct pointer type generated from proto
+	// to avoid nil in neutron, null in json and rust marshalling errors, here we initialize it with "default" values
+	// which is not equal 0 because github.com/cosmos/cosmos-sdk/codec/types skips true default values
+	emptyHeight := ibcclienttypes.NewHeight(1, 1)
+	if query.LastSubmittedResultRemoteHeight == nil {
```

Comment:
Why not just to use `Option<u64>` on sc level?

Reply:
if you meant `Option<Height>`,  I believe couple of simple lines in neutron node is a bit preferable than unnecessary complications in our smart contracts, also we should init this struct anyway (in other place), even w `Option`